core/types/schema.go

```diff
@@ -278,7 +278,19 @@ type Attribute struct {
 	Value string        `json:"value,omitempty"`
 }
 
-func (a *Attribute) Clean() error {
+// Clean validates rules about the data in the struct (naming conventions, syntax, etc.).
+func (a *Attribute) Clean(col *Column) error {
+	switch a.Type {
+	case MIN, MAX:
+		if !col.Type.Equals(IntType) {
+			return fmt.Errorf("attribute %s is only valid for int columns", a.Type)
+		}
+	case MIN_LENGTH, MAX_LENGTH:
+		if !col.Type.Equals(TextType) {
+			return fmt.Errorf("attribute %s is only valid for text columns", a.Type)
+		}
+	}
+
```

Comment:
This is good. 
Maybe also add index and foreign_key validation? i.e. make sure referred column exists. 
That's so far the validations I can think of that we can do for table definition.